backend/src/bundles/achievements/achievement.repository.ts

```diff
@@ -0,0 +1,79 @@
+import { AchievementEntity } from '~/bundles/achievements/achievement.entity.js';
+import { type AchievementModel } from '~/bundles/achievements/achievement.model.js';
+import { type Repository } from '~/common/types/types.js';
+
+class AchievementRepository implements Repository {
+    private achievementModel: typeof AchievementModel;
+
+    public constructor(achievementModel: typeof AchievementModel) {
+        this.achievementModel = achievementModel;
+    }
+
+    public async find(
+        query: Record<string, unknown>,
+    ): Promise<AchievementEntity | null> {
+        const achievement = await this.achievementModel
+            .query()
+            .findOne(query)
+            .execute();
+
+        if (!achievement) {
+            return null;
+        }
+
+        return AchievementEntity.initialize(achievement);
+    }
+
+    public async findAll(): Promise<AchievementEntity[]> {
+        const achievements = await this.achievementModel.query().execute();
+
+        return achievements.map((achievement) => {
+            return AchievementEntity.initialize(achievement);
+        });
+    }
+
+    public async findById(id: number): Promise<AchievementEntity | null> {
+        const achievement = await this.achievementModel
+            .query()
+            .findById(id)
+            .execute();
+        return achievement ? AchievementEntity.initialize(achievement) : null;
+    }
+
+    public async findByUserId(
+        userId: number,
+    ): Promise<AchievementEntity[] | null> {
+        const achievements = await this.achievementModel
+            .query()
+            .joinRelated('userAchievements')
+            .where('userAchievements.user_id', userId)
+            .select('achievements.*', 'userAchievements.created_at');
```

Comment:
you don't need to user snake_case
Also I'm curious if there is no ambiguity in request since achievements and userAchievements both have createdAt fieild

Reply:
Here I want to clarify.

createdAt for Achievements is the date when the achievement was created.

createdAt for userAchievements is the date when the user received the achievement

Therefore, these dates are different.

![Screenshot_created1](https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/assets/106997950/a58f93f6-d4d2-4c40-b8fd-8867dc46e948)

![Screenshot_created2](https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/assets/106997950/1ad5d6e2-e63c-4d9b-8f11-9eba58e8f1f0)